src/server.js

```diff
@@ -0,0 +1,96 @@
+'use strict';
+
+const http = require('http');
+const path = require('path');
+const fs = require('fs');
+const zlib = require('zlib');
+const multiparty = require('multiparty');
+
+const server = http.createServer((req, res) => {
+  if (req.url === '/') {
+    const pathFileIndexHTML = path.join(__dirname, 'index.html');
+
+    const readStream = fs.createReadStream(pathFileIndexHTML);
+
+    res.statusCode = 200;
+    res.setHeader('Content-Type', 'text/html');
+
+    readStream.pipe(res);
+
+    readStream.on('error', (err) => {
+      res.statusCode = 404;
+      res.end(err);
+    });
+
+    return;
+  }
+
+  if (req.url === '/compress') {
+    const form = new multiparty.Form();
+
+    form.parse(req, (err, fields, files) => {
+      if (err) {
+        res.statusCode = 404;
+        res.end('Error');
+
+        return;
+      }
+
+      const file = files.file[0];
+      const compress = fields.compress[0];
+
+      const fileName = file.originalFilename.split('.')[0];
+
+      const readableStream = fs.createReadStream(file.path);
+
+      let compressionStream;
+
+      switch (compress) {
+        case 'gzip':
+          res.setHeader(
+            'Content-Disposition',
+            `attachment; filename=${fileName}.gz`,
+          );
+          compressionStream = zlib.createGzip();
+          break;
+        case 'deflate':
+          res.setHeader(
+            'Content-Disposition',
+            `attachment; filename=${fileName}.deflate`,
+          );
+          compressionStream = zlib.createDeflate();
+          break;
+        case 'br':
+          res.setHeader(
+            'Content-Disposition',
+            `attachment; filename=${fileName}.br`,
+          );
+          compressionStream = zlib.createBrotliCompress();
+          break;
+        default:
+          res.statusCode = 400;
+          res.end('Invalid compression type');
+
+          return;
+      }
```

Comment:
Now everything is written correctly, but since most of the part for the `setHeader` is the same and only the extension changes, it would be possible to choose the extension itself with a `switch...case`, and to register the installation of the header once:
```
let compressionExtension;
switch (compress) {
  case 'gzip':
    compressionExtension = 'gz';
    compressionStream = zlib.createGzip();
      break;
  case 'deflate':
    compressionExtension = 'deflate';
    compressionStream = zlib.createDeflate();
      break;
  case 'br':
    compressionExtension = 'br';
    compressionStream = zlib.createBrotliCompress();
      break;
    ...
}
res.setHeader(
            'Content-Disposition',
            `attachment; filename=${fileName}.${compressionExtension}`,
          );
```
This will reduce the amount of code written and improve readability. Also, such checks could be transferred to a separate helper function, for even better readability:
```
function getCompressionParams(compress) {
    ...
    return { compressionExtension, compressionStream  };
}
...
const { compressionExtension, compressionStream } = getCompressionParams(compress);
```